extensions/amp-story/0.1/variable-service.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {StateChangeType} from './navigation-state';
+import {dev} from '../../../src/log';
+
+
+/**
+ * Variable service for amp-story.
+ * Used for URL replacement service. See usage in src/url-replacements-impl.
```

Comment:
Has the PR for usage in url-replacements-impl been created yet? I'm curious if we should be extending `VariableSource` here like `A4AVariableSource` does.

Reply:
I think that would be a misnomer, as `VariableSource` refers to the global provider that is instantiated by the runtime. 

The PR for `url-replacements-impl` has not been created, but you can [see usage here](https://github.com/newmuis/amphtml-story/blob/master/src/service/url-replacements-impl.js#L468).

Reply:
I see, thanks. 

I meant that an alternative is extending `VariableSource` and wrapping `GlobalVariableSource` internally, and creating a custom `UrlReplacements`. This way we won't pollute the `GlobalVariableSource` URL namespace with extension-specific variables. This is fine for now though something to consider later.